### ISSUE TITLE
Rule execution/availability in .Net Framework vs. .NET

### ISSUE THREAD

**Issue by N-Olbert** [dotnet/roslyn-analyzers#6045]
### Analyzer

**Diagnostic ID**: [CA1024](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1024): `Use properties where appropriate`
 (for example, this affects a lot of other analyzers too)

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 6.0.0 (Latest)

### Describe the bug

Within a .Net Framework (legacy) application some rules does not seem to get validated (f. e. CA1024).
Is this intended? I understand that some rules don't make sense for .NET Framework (f. e. CA1307, since most overloads don't exist in 4.8), but I can’t see this limitation f. e. for CA1024.
If it is indeed intended: is there a list where one can look up which rules are enables within .Net Framework / .NET?

### Steps To Reproduce

For example, take the following class:

```
    internal class Bar
    {
        public string GetFoo()
        {
            return "foo" + this.ToString();
        }
    }
```
Set all analyzers to active in your .editorconfig: 
`dotnet_analyzer_diagnostic.severity = error`

### Expected behavior
CA1024 should raise (?)

### Actual behavior
If you compile the class within a .NET 6-project, CA1024 is raised.
If you compile the class within a .Net Framework-project, CA1024 is NOT raised.
